[2023-12-22T19:37:10.574+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: batch_layer.income_statement manual__2023-12-22T19:32:04.625550+00:00 [queued]>
[2023-12-22T19:37:10.642+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: batch_layer.income_statement manual__2023-12-22T19:32:04.625550+00:00 [queued]>
[2023-12-22T19:37:10.643+0000] {taskinstance.py:2171} INFO - Starting attempt 2 of 2
[2023-12-22T19:37:10.655+0000] {taskinstance.py:2192} INFO - Executing <Task(SparkSubmitOperator): income_statement> on 2023-12-22 19:32:04.625550+00:00
[2023-12-22T19:37:10.659+0000] {standard_task_runner.py:60} INFO - Started process 333 to run task
[2023-12-22T19:37:10.661+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'batch_layer', 'income_statement', 'manual__2023-12-22T19:32:04.625550+00:00', '--job-id', '9', '--raw', '--subdir', 'DAGS_FOLDER/BatchLayerDag.py', '--cfg-path', '/tmp/tmpecz3y80w']
[2023-12-22T19:37:10.663+0000] {standard_task_runner.py:88} INFO - Job 9: Subtask income_statement
[2023-12-22T19:37:10.708+0000] {task_command.py:423} INFO - Running <TaskInstance: batch_layer.income_statement manual__2023-12-22T19:32:04.625550+00:00 [running]> on host 1e7ecb9f94ff
[2023-12-22T19:37:10.803+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='airflow@example.com' AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='batch_layer' AIRFLOW_CTX_TASK_ID='income_statement' AIRFLOW_CTX_EXECUTION_DATE='2023-12-22T19:32:04.625550+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2023-12-22T19:32:04.625550+00:00'
[2023-12-22T19:37:10.808+0000] {spark_submit.py:223} INFO - Could not load connection string spark-conn, defaulting to yarn
[2023-12-22T19:37:10.810+0000] {spark_submit.py:351} INFO - Spark-Submit cmd: spark-submit --master yarn --name arrow-spark /opt/bitnami/spark/jobs/dags/batch_proc_0.py
[2023-12-22T19:37:10.818+0000] {spark_submit.py:521} INFO - /home/airflow/.local/bin/spark-submit: line 27: /opt/spark/bin/spark-class: No such file or directory
[2023-12-22T19:37:10.823+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 160, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 452, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --name arrow-spark /opt/bitnami/spark/jobs/dags/batch_proc_0.py. Error code is: 127.
[2023-12-22T19:37:10.830+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=batch_layer, task_id=income_statement, execution_date=20231222T193204, start_date=20231222T193710, end_date=20231222T193710
[2023-12-22T19:37:10.847+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 9 for task income_statement (Cannot execute: spark-submit --master yarn --name arrow-spark /opt/bitnami/spark/jobs/dags/batch_proc_0.py. Error code is: 127.; 333)
[2023-12-22T19:37:10.874+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2023-12-22T19:37:10.883+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
